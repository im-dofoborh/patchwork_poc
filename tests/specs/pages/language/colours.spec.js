import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import ColoursPage from '~/pages/language/colours';
import { site, theme, colours } from '~/tests/fixtures';

describe('Colours page', () => {
  let $fabric = {
    getColours: async () => {
      return {
        data: colours,
      }
    }
  };

  let localVue = null;
  let wrapper = null;
  let store = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('colours-page', ColoursPage);
    localVue.use(Vuex);
    store = new Vuex.Store({
      state: {
        currentSite: site,
        currentTheme: theme,
      },
    });
    wrapper = mount(ColoursPage, {
      localVue,
      store,
      mocks: {
        $fabric
      },
    });
  })

  it('renders', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('loads asyncData', async () => {
    const app = wrapper.vm;
    const asyncData = await wrapper.vm.$options.asyncData({ app });
    expect(asyncData).toEqual({ colours });
  });

  it('renders with loaded data', () => {
    wrapper = mount(ColoursPage, {
      localVue,
      store,
      data() {
        return {
          colours,
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
