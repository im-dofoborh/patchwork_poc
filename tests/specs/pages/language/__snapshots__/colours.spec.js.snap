// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colours page renders 1`] = `
<section class="sg-colours-page">
  <div class="sg-content container">
    <div class="sg-content__header">
      <h1 class="heading-1">Colours</h1>
    </div>
    <div class="sg-content__section">
      <div class="editor-content">
        <p>
          These are all of the colours currently declared in the
          Slate color.scss theme. It does not mean that
          all of these are in use. You will also have access to all of the
          colours in the parent theme, provided that you have not overridden
          them.
        </p>
      </div>
      <div class="sg-sample-grid mt-lg"></div>
    </div>
  </div>
</section>
`;

exports[`Colours page renders with loaded data 1`] = `
<section class="sg-colours-page">
  <div class="sg-content container">
    <div class="sg-content__header">
      <h1 class="heading-1">Colours</h1>
    </div>
    <div class="sg-content__section">
      <div class="editor-content">
        <p>
          These are all of the colours currently declared in the
          Slate color.scss theme. It does not mean that
          all of these are in use. You will also have access to all of the
          colours in the parent theme, provided that you have not overridden
          them.
        </p>
      </div>
      <div class="sg-sample-grid mt-lg">
        <div>
          <div class="sg-colours-page__palette">
            <div class="sg-colours-page__palette-swatch" style="background-color: rgb(33, 189, 255);"></div>
            <div class="sg-colours-page__palette-name"><code>$color--highlight--light</code></div>
            <div class="sg-colours-page__palette-hex"><code>#21bdff</code></div>
            <!---->
            <!---->
          </div>
        </div>
        <div>
          <div class="sg-colours-page__palette">
            <div class="sg-colours-page__palette-swatch" style="background-color: rgb(0, 123, 182);"></div>
            <div class="sg-colours-page__palette-name"><code>$color--highlight--regular</code></div>
            <div class="sg-colours-page__palette-hex"><code>#007bb6</code></div>
            <!---->
            <!---->
          </div>
        </div>
        <div>
          <div class="sg-colours-page__palette">
            <div class="sg-colours-page__palette-swatch" style="background-color: rgb(0, 89, 130);"></div>
            <div class="sg-colours-page__palette-name"><code>$color--highlight--dark</code></div>
            <div class="sg-colours-page__palette-hex"><code>#005982</code></div>
            <div class="sg-colours-page__palette-note">
              <p class="sg-colours-page__palette-note-text">From:</p> <code>
                #000
              </code>
            </div>
            <div class="sg-colours-page__palette-note">
              <p class="sg-colours-page__palette-note-text">
                These comments are probably never used
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
